### PR TITLE
AutoComplete-add-separator-to-style-itemLabel

### LINKED ISSUE #10489: AutoComplete: Enable styling of the selected items by a delimiter
### Description

I find a good (client side) solution to enable the styling of the selected items in `AutoComplete`.  

If the autoComplete `itemLabel` contains a line break (= `/\n`), then it will be replaced by a `span`-element so that the user is able to style each token within the selected item. For example: `itemLabel="a /\n b /\n c"` will be replaced by 

```
<span class="ui-autocomplete-token-label-1">a</span>
<span class="ui-autocomplete-token-label-2">b</span>
<span class="ui-autocomplete-token-label-3">c</span>
```

I extend `PrimeFaces.widget.AutoComplete.prototype.addItem` to make these changes (look into the reproducer (`webapp/resources/test.js`). 

**Look at the reproducer:**

[primefaces-test-jakarta.zip](https://github.com/primefaces/primefaces/files/12333896/primefaces-test-jakarta.zip)

### Additional context

After this issue is resolved, I will provide a PR so that we can style the selected items by simply putting line break (e.g. "/\n") as a delimiter. This PR will also solve:

- https://github.com/primefaces/primefaces/issues/10470
- https://github.com/primefaces/primefaces/issues/10411
- https://github.com/primefaces/primefaces/issues/1903

since the user is now able to style the selectable elements based on their `span` elements.

### PR DESCRIPTION
**Pull request by nimo23**
Fixes https://github.com/primefaces/primefaces/issues/10489.

### CODE REVIEW

File: primefaces/src/main/resources/META-INF/resources/primefaces/autocomplete/autocomplete.js [hunk 1]
```diff
@@ -1185,13 +1186,24 @@ PrimeFaces.widget.AutoComplete = PrimeFaces.widget.BaseWidget.extend({
 
                 var itemLabelEscaped = PrimeFaces.escapeHTML(itemLabel);
                 var itemValueEscaped = PrimeFaces.escapeHTML(itemValue);
+                
+                var itemLabelHtml = null;
+                var separator = $this.cfg.separator;
+                if (itemLabelEscaped.includes(separator)) {
```
Comment (melloware) [inline primefaces/src/main/resources/META-INF/resources/primefaces/autocomplete/autocomplete.js]:
```suggestion
                if (separator && itemLabelEscaped.includes(separator)) {
```

File: primefaces/src/main/resources/META-INF/resources/primefaces/autocomplete/autocomplete.js [hunk 2]
```diff
@@ -1185,13 +1186,24 @@ PrimeFaces.widget.AutoComplete = PrimeFaces.widget.BaseWidget.extend({
...
+                }
+
                 var itemDisplayMarkup = '<li data-token-value="' + itemValueEscaped;
                 itemDisplayMarkup += '"class="ui-autocomplete-token ui-state-active ui-corner-all ui-helper-hidden';
                 itemDisplayMarkup += (itemStyleClass === '' ? '' : ' ' + itemStyleClass) + '" '
                 itemDisplayMarkup += 'role="option" aria-label="' + itemLabelEscaped + '" ';
                 itemDisplayMarkup += 'aria-selected="true">';
                 itemDisplayMarkup += '<span class="ui-autocomplete-token-icon ui-icon ui-icon-close" aria-hidden="true"></span>';
-                itemDisplayMarkup += '<span class="ui-autocomplete-token-label">' + itemLabelEscaped + '</span></li>';
+                itemDisplayMarkup += '<span class="ui-autocomplete-token-label">' + (itemLabelHtml || itemLabelEscaped) + '</span></li>';
```
Comment (melloware) [inline primefaces/src/main/resources/META-INF/resources/primefaces/autocomplete/autocomplete.js]: I think we need Integration tests added proving this case works.

**Review by melloware — CHANGES_REQUESTED** [review]

File: primefaces/src/main/resources/META-INF/resources/primefaces/autocomplete/autocomplete.js (hunk 2 above)
Comment (nimo23) [reply primefaces/src/main/resources/META-INF/resources/primefaces/autocomplete/autocomplete.js]: I don't know how to do integration tests. But I can also use the following version which would elminate the need of the integration test, or?:

```
 itemDisplayMarkup += '<span class="ui-autocomplete-token-label">' + (itemLabelHtml == null ? itemLabelEscaped : itemLabelHtml) + '</span></li>';
```

Comment (nimo23) [reply primefaces/src/main/resources/META-INF/resources/primefaces/autocomplete/autocomplete.js]: I changed the code which does not need an integration test:

`itemDisplayMarkup += '<span class="ui-autocomplete-token-label">' + (itemLabelHtml == null ? itemLabelEscaped : itemLabelHtml) + '</span></li>';`

**Comment by tandraschko** [comment]
see my last comment in the issue

**Pull request closed without merging**